From ecc0714d0c132acdce1d7363567917e00ece26f9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20=C4=8Cerm=C3=A1k?= <sairon@sairon.cz>
Date: Tue, 13 May 2025 17:19:59 +0200
Subject: [PATCH] rpi-eeprom-update: adjust hassos-boot discovery
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Make discovery of boot partition compatible with HAOS without the need
to set BOOTFS in /etc/default/rpi-eeprom-update.

Signed-off-by: Jan Čermák <sairon@sairon.cz>
Upstream: not applicable
---
 rpi-eeprom-update | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rpi-eeprom-update b/rpi-eeprom-update
index b970142..720efb7 100755
--- a/rpi-eeprom-update
+++ b/rpi-eeprom-update
@@ -725,7 +725,7 @@ findBootFS()
    # If ${BOOTFS} is not writable OR is not on /dev/mmcblk0 then error because the ROM
    # can only load recovery.bin from the on-board SD-CARD slot or the EEPROM.
 
-   if blkid | grep -qE "/dev/mmcblk0p1.*LABEL_FATBOOT.*RECOVERY.*TYPE.*vfat"; then
+   if blkid | grep -qE "/dev/mmcblk0p1.*LABEL_FATBOOT.*hassos-boot.*TYPE.*vfat"; then
       TMP_BOOTFS_MNT="$(mktemp -d)"
       mount /dev/mmcblk0p1 "${TMP_BOOTFS_MNT}"
       BOOTFS="${TMP_BOOTFS_MNT}"
